db/crud.go

```diff
@@ -869,6 +869,33 @@ func (db *DatabaseCollectionWithUser) OnDemandImportForWrite(ctx context.Context
 	return nil
 }
 
+// updateHLV updates the HLV in the sync data appropriately based on what type of document update event we are encountering
+func (db *DatabaseCollectionWithUser) updateHLV(d *Document, docUpdateEvent uint32) (*Document, error) {
```

Comment:
Once you've updated the enum (see other note), should make this change:
```suggestion
func (db *DatabaseCollectionWithUser) updateHLV(d *Document, docUpdateEvent docUpdateEventType) (*Document, error) {
```